design-proposals/sriov-interfaces-mapping.md

```diff
@@ -0,0 +1,694 @@
+| VM creation/restart/migration<br/>(new virt-launcher) | 3. Regression<br/>Virt-handler sends the network-to-pci mapping information to virt-launcher through gRPC, virt-launcher should fall back to legacy mapping method. | 4. No regression                                                                                                                                                                                                                                                        |
+
+#### Conclusion
+- (2) and (3):<br/>
+There might be a disruption to VMs when they were created:
+  - After the virt-controller is updated (new virt-launcher), on a node with an old instance of virt-handler.
+  - Before the virt-controller is updated (old virt-launcher), on a node with a new instance of virt-handler.
+  <br/><br/>
+  Since the gRPC command server (virt-launcher) and client (virt-handler) runs with different API versions, virt-launcher may fail to synchronize with the new VMI state.
+  <br/><br/>
+
+- (4) When both virt-handler and virt-controller are updated, new VMs will be created with the correct SR-IOV interfaces assignment.
+
+### Pros
+1. No need to involve virt-controller.
+2. No need to create annotations on the pod or VMI. 
+
+### Cons
+1. Requires API changes to virt-handler to virt-launcher commands API [[7]](https://github.com/kubevirt/kubevirt/blob/1a8f08103e48c5f2bb2f5826d118507ce7ec1f0c/pkg/handler-launcher-com/cmd/v1/cmd.pb.go) [[8]](https://github.com/kubevirt/kubevirt/blob/1a8f08103e48c5f2bb2f5826d118507ce7ec1f0c/pkg/handler-launcher-com/cmd/v1/cmd.pb.go#L334).
+2. virt-handler needs to fetch the pod object in order to get the mapping information.
+
+## 4. Expose network-to-pci mapping through the VMI status
+This option uses the same basic ideas from [option 2](#2.virt-controller-exposes-network-to-pci-mapping-through-two-VMI-annotations) but instead of using VMI annotations it
+uses VMI Status additional fields to persist the mapping.
+
+1. Extend the VMI status API to reflect each SR-IOV interface underlying PCI address.
+
+2. The virt-controller updates the status fields with the mapping information.
+
+3. virt-launcher VMI extract the network-to-pci mapping from the VMI status and create each HostDevice accordingly.<br/>
+   In case there is not enough information in the VMI status (i.e: missing host PCI address) fall back to legacy mapping method. 
+
+#### Pros
+1. No need for annotations.
+
+#### Cons
+1. The status field will have to support mapping from two pods, supporting migration.
```

Comment:
Yes that is a one-time action, where luckily most entrypoints already exist since we already have to cleanup the status after migrations. Then we are in the good situation that we don't have to think about separate flows anymore.